### ISSUE TITLE
Get Travis to build Toga

### ISSUE THREAD

**Issue by 1ethanhansen** [Zynh0722/ProjectCodeName#60]
Need to actually run the program to test on Travis, but also need a way to stop testing because it is a while loop

**Comment by 1ethanhansen** [comment]
It builds just fine, my issue is with getting it to stop running programatically and testing individual toga functions because they require fancy toga object inputs that we don't have. An option is to make all of our functions do the stuff we want sans fancy inputs, then call those functions from toga specific functions. Kind of clunky in code but might be what we need

**Comment by 1ethanhansen** [comment]
nvm to the whole
> It builds just fine

thing.
Output from Travis:
```$ python trackmanagementsystem/__main__.py
Unable to init server: Could not connect: Connection refused
Unable to init server: Could not connect: Connection refused
/home/travis/.travis/functions: line 109:  6740 Segmentation fault      (core dumped) python trackmanagementsystem/__main__.py
The command "python trackmanagementsystem/__main__.py" exited with 139.
```

**Comment by 1ethanhansen** [comment]
That `unable to init server` is talking about travis being command line only and not being allowed to build graphics

**Comment by 1ethanhansen** [comment]
We have a few courses of action:

1. I will post on a couple forums asking about anyone running toga apps on Travis
2. Try some suggestions from those forums
3. Just do unit tests on Travis, leaving the integration tests to us to run manually

Hopefully we don't have to resort to 3, but if need be that wouldn't be the end of the world

**Comment by 1ethanhansen** [comment]
From a Maintainer on [the beeware gitter](https://gitter.im/beeware/general/#):

> @1ethanhansen_gitlab Hi! And congratulations - you’ve just found the one seriously hard problem with building GUI apps: Testing
Testing GUIs is a hard problem - because GUIs aren’t naturally suited to automated testing.
This is something Toga itself faces - how do you build tests for a GUI framework when the thing you’re building is all about GUI appearance?
Unfortunately, I don’t have any good answers here at the moment. Testing GUIS is just hard - and usually involves a manual compliance suite (i.e. ,write a script: “open the app, click on X, scroll down to Y, check that the value is 15”) and have someone run it whenever you want to test the app. It’s not automated, but it’s a test.
It is possible to automate some GUIs - pyautogui is a good example of the sort of thing that can be done - but that’s a long way from an integration testing framework.

In other words, we might need to resort to 3

**Comment by 1ethanhansen** [comment]
Decision made to not use Travis for integration tests, closing this out. We will do manual tests
